### PR TITLE
redux comments DRAFT WIP

### PR DESCRIPTION
**Pull request by jessopb**
Adds comment functionality to redux

### CODE REVIEW

File: src/redux/actions/comments.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const claim = selectClaimsByUri(state)[uri];
+    const claimId = claim ? claim.claim_id : null;
+
+    dispatch({
+      type: ACTIONS.COMMENT_LIST_STARTED,
+    });
+    Lbry.comment_list({
+      claim_id: claimId,
+    })
+      .then(results => {
```
Comment (seanyesmunt) [inline src/redux/actions/comments.js]: This is a `CommentListResponst` type

File: src/redux/actions/comments.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        dispatch({
+          type: ACTIONS.COMMENT_LIST_COMPLETED,
+          data: {
+            comments: results,
+            claimId: claimId,
+            uri: uri,
+          },
+        });
+      })
+      .catch(error => {
```
Comment (seanyesmunt) [inline src/redux/actions/comments.js]: We should have a `COMMENT_LIST_FAILED` action that handles this error. (we don't need to display the error in the ui right now)

File: src/redux/actions/comments.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      channel_id,
+    })
+      .then((result: Comment) => {
+        dispatch({
+          type: ACTIONS.COMMENT_CREATE_COMPLETED,
+          data: {
+            response: result,
+          },
+        });
+        dispatch({
```
Comment (seanyesmunt) [inline src/redux/actions/comments.js]: Whatever happens in the `COMMENT_LIST_UPDATED` handler can just be done in `COMMENT_CREATE_COMPLETED`.

Just append the new comment there, then we don't need two actions.

File: src/redux/reducers/comments.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as ACTIONS from 'constants/action_types';
+
+// TODO change to handleActions()
```
Comment (seanyesmunt) [inline src/redux/reducers/comments.js]: 🕺

File: src/redux/reducers/comments.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  isLoading: false,
+};
+
+reducers[ACTIONS.COMMENT_CREATE_STARTED] = (state: CommentsState, action: any): CommentsState =>
+  Object.assign({}, state, {
+    isLoading: true,
+  });
+
+reducers[ACTIONS.COMMENT_CREATE_FAILED] = (state: CommentsState, action: any): CommentsState => {
+  // TODO: handle error.. what else?
```
Comment (seanyesmunt) [inline src/redux/reducers/comments.js]: is `isLoading` to `false`

File: src/redux/reducers/comments.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  return state;
+};
+
+reducers[ACTIONS.COMMENT_CREATE_COMPLETED] = (state: CommentsState, action: any): CommentsState => {
+  const { comments }: any = action.data;
```
Comment (seanyesmunt) [inline src/redux/reducers/comments.js]: Add this new comment to the list of comments here.

File: dist/bundle.es.js [hunk 7]
```diff
@@ -2769,6 +2778,74 @@ const doDeleteTag = name => ({
...
+}
+
+function doCommentCreate(comment = '', claim_id = '', channel, parent_id) {
+  return (dispatch, getState) => {
+    const state = getState();
+    dispatch({
+      type: COMMENT_CREATE_STARTED
+    });
+    const myChannels = selectMyChannelClaims(state);
+    const namedChannelClaim = myChannels.find(myChannel => myChannel.name === channel);
```
Comment (osilkin98) [inline dist/bundle.es.js]: The SDK already does this anyway when signing the comments

**Pull request merged**